feat: add support for object input

michaelfaith opened the pull request:
<!-- 👋 Hi, thanks for sending a PR to package-json-validator! 💖.
Please fill out all fields below and make sure each item is true and [x] checked.
Otherwise we may not be able to review your PR. -->

## PR Checklist

- [x] Addresses an existing open issue: fixes #66
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/package-json-validator/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/package-json-validator/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

This change introduces the ability to pass an object to the validate function in addition to a stringified json object.

I updated the docs and added additional unit tests to verify every form of validation we're already testing but with object input instead of string.

![screenshot of tests passing for new unit tests](https://github.com/user-attachments/assets/e0f2ea34-f246-409b-9852-e20ad01caec2)

**_NOTE:_** As far as the order of things, it would probably be best to land https://github.com/JoshuaKGoldberg/package-json-validator/pull/135 before this one, and then let me update this change to use `validate` directly in the new unit tests before merging this.

Closes #66

JoshuaKGoldberg commented on README.md:
````diff
@@ -73,7 +73,7 @@ const { PJV } = require("package-json-validator");
 ```
 
-Example1:
+Example:
````
[Docs] Nit: to align with the later one...

```suggestion
Example using a string:
```

JoshuaKGoldberg commented on README.md:
````diff
@@ -138,6 +138,19 @@ console.log(data);
 }
 ```
 
+Example using an object:
````
[Question] I feel like the object use case is probably a little more real-world than the string one. WDYT about putting it first?

It is nice how this one is a lot fewer lines than the string one. Which IMO makes it even better for going first...

Not a blocker on my end, I'm happy either way.

michaelfaith replied:
Updated

JoshuaKGoldberg commented on src/PJV.ts:
```diff
@@ -212,11 +212,16 @@ type ValidationOutput = {
 	options: ValidationOptions = {},
 ): ValidationOutput => {
-	const parsed = parse(data);
+	let parsed;
+	if (typeof data !== "object") {
+		parsed = parse(data);
+	} else {
+		parsed = data;
+	}
```
[Style] Nitpicking, no need for this to be a `let` (https://github.com/sindresorhus/eslint-plugin-unicorn/blob/e5e17f8edd7bee0b375313c3708955980cbbfe96/docs/rules/prefer-ternary.md):

```suggestion
	const parsed = typeof data === "object" ? data : parse(data);
```

JoshuaKGoldberg reviewed (approved):
🔥 Awesome! Nice to see such a straightforward docs & implementation change - and thorough tests alongside it.

Just a few suggestions, nothing blocking from me. I'm happy whenever you want to merge this. ✅ Thanks!

michaelfaith commented:
I addressed all the feedback.  Like I mentioned in the PR Desc, though, https://github.com/JoshuaKGoldberg/package-json-validator/issues/124 should probably land first, and then I'll update this PR to change the tests to use `validate` directly, before merging this one.

michaelfaith commented:
Rebased to resolve conflicts, and should be ready to go
